### PR TITLE
Add decoder serialization

### PR DESCRIPTION
**Pull request by mikeyshulman**
RFC

### CODE REVIEW

File: pyctcdecode/decoder.py [hunk 1]
```diff
@@ -665,6 +668,65 @@ def decode_batch(
         decoded_text_list: List[str] = pool.map(p_decode, logits_list)
         return decoded_text_list
 
+    def save_to_dir(self, filepath: str) -> None:
+        """Save a decoder to a directory."""
+        alphabet_path = os.path.join(filepath, self._ALPHABET_SERIALIZED_FILENAME)
+        with open(alphabet_path, "wb") as fi:
+            pickle.dump(self._alphabet, fi)  # nosec
```
Comment (mikeyshulman) [inline pyctcdecode/decoder.py:675]: we could decide to serialize this to plain text- the alphabet is just a list of labels and whether its BPE. Looking for some comments

File: pyctcdecode/decoder.py [hunk 2]
```diff
@@ -665,6 +667,66 @@ def decode_batch(
...
+    @classmethod
+    def load_from_dir(cls, filepath: str) -> "BeamSearchDecoderCTC":
+        """Load a decoder from a directory."""
+        filenames = cls.parse_directory_contents(filepath)
+        with open(filenames["alphabet"], "r") as fi:  # type: ignore
+            alphabet = Alphabet.loads(fi.read())
+        if filenames["language_model"] is None:
+            language_model = None
+        else:
+            print(filenames)
```
Comment (gkucsko) [inline pyctcdecode/decoder.py:726]: remove?

**Review by gkucsko — COMMENTED** [review]
lgtm

**Review by gkucsko — APPROVED** [review]
